org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/ReferenceBinding.java

```diff
@@ -1337,6 +1338,24 @@ public boolean implementsInterface(ReferenceBinding anInterface, boolean searchH
 			}
 		}
 	}
+	// see https://github.com/eclipse-jdt/eclipse.jdt.core/issues/629
```

Comment:
I think this solves the issue, But I doubt why superInterfaces are not resolved when the null annotation processing is enabled. Its something I spent last 2-3 days few hours each to see if I could 😞. 

But unless we get some input on why it's not properly resolved, This will be a good quick fix for the issue. We can change later if we find a better fix suggestion.